### PR TITLE
[GPU] dump graphs during execution

### PR DESCRIPTION
**Pull request by sungeunk**
`-filenaming: cldnn_program_exec_{network_id}_{program_id}_{iter}.graph`
-delete filter

### Tickets:
 - 125771

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/network.cpp [hunk 1]
```diff
@@ -1121,6 +1122,22 @@ void network::execute_impl(const std::vector<event::ptr>& events) {
...
 
+    GPU_DEBUG_IF(!debug_config->dump_graphs.empty() && debug_config->is_target_iteration(curr_iter)) {
+        auto get_fixed_str = [](int value, int length = 2) -> std::string {
+            std::ostringstream ss;
+            ss << std::setw(length) << std::setfill('0') << std::to_string(value);
+            return ss.str();
+        };
+        std::string path = get_dir_path(get_config());
+        if (!path.empty()) {
+            std::ofstream ofs(path + "cldnn_program_exec_" + get_fixed_str(get_id()) + "_" + get_fixed_str(get_program()->get_id())
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/network.cpp:1133]: Something like `0_0_0` would be difficult to understand. Could you make it something like `p0_n0_0`?

**Review by isanghao — COMMENTED** [review]
Looks good to me in general. Left minor comment.

File: src/plugins/intel_gpu/src/graph/network.cpp [hunk 2]
```diff
@@ -1121,6 +1122,22 @@ void network::execute_impl(const std::vector<event::ptr>& events) {
                        << data_shape_str.str() << std::endl;
     }
 
+    GPU_DEBUG_IF(!debug_config->dump_graphs.empty() && debug_config->is_target_iteration(curr_iter)) {
```
[the remark this answers is not in the document]
Comment (sungeunk) [reply src/plugins/intel_gpu/src/graph/network.cpp:1125]: Dump iter for inner network can be implemented via other task(PR).

File: src/plugins/intel_gpu/src/graph/program.cpp [hunk 3]
```diff
@@ -1217,24 +1217,22 @@ void program::remove_nodes(std::vector<program_node*>& to_remove) {
...
-                           std::function<bool(program_node const&)> const& filter) const {
+void program::dump_program(const char* stage, bool with_full_info) const {
     std::string path = get_dir_path(_config);
     if (path.empty() || !with_full_info) {
         return;
     }
 
-    std::ofstream graph(path + "cldnn_program_" + std::to_string(prog_id) + "_" + stage + ".graph");
-    dump_graph_init(graph, *this, filter);
+    std::ofstream graph(path + "cldnn_program_p" + std::to_string(prog_id) + "_" + stage + ".graph");
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/program.cpp:1226]: you don't need to change name here. It already says "program"...

**Review by isanghao — APPROVED** [review]

**Pull request merged**